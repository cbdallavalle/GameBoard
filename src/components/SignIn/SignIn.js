import React from 'react';
import PropTypes from 'prop-types';
import './SignIn.css';

export const SignIn = ({ handleChange, email, passwordOne }) => {
  return (
    <div className="SignIn">
      <input 
        type="email"
        placeholder="email"
        name="email"
        onChange={ (event) => handleChange(event) }
        value={ email }
      />
      <input 
        type="password"
        placeholder="Password"
        name="passwordOne"
        onChange={ (event) => handleChange(event) }
        value={ passwordOne }
      />
    </div>
  )
}


SignIn.propTypes = {
  handleChange: PropTypes.func.isRequired,
  email: PropTypes.string.isRequired,
  passwordOne: PropTypes.string.isRequired
};