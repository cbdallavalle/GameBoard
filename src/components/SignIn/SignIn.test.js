import React from 'react';
import { shallow } from 'enzyme';
import { SignIn } from './SignIn';

describe("SignIn", () => {
  const mockHandleChange = jest.fn();

  it('should match the snapshot', () => {
    const wrapper = shallow(
      <SignIn 
        handleChange={mockHandleChange}
        email={'email'}
        passwordOne={'password'}
      />
    )

    expect(wrapper).toMatchSnapshot();
  })
})