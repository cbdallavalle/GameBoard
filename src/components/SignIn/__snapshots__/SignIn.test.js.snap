// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    email="email"
    handleChange={[Function]}
    passwordOne="password"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          name="email"
          onChange={[Function]}
          placeholder="email"
          type="email"
          value="email"
/>,
        <input
          name="passwordOne"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value="password"
/>,
      ],
      "className": "SignIn",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "email",
          "onChange": [Function],
          "placeholder": "email",
          "type": "email",
          "value": "email",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "passwordOne",
          "onChange": [Function],
          "placeholder": "Password",
          "type": "password",
          "value": "password",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            name="email"
            onChange={[Function]}
            placeholder="email"
            type="email"
            value="email"
/>,
          <input
            name="passwordOne"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value="password"
/>,
        ],
        "className": "SignIn",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "email",
            "onChange": [Function],
            "placeholder": "email",
            "type": "email",
            "value": "email",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "passwordOne",
            "onChange": [Function],
            "placeholder": "Password",
            "type": "password",
            "value": "password",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
